crates/services/p2p/src/discovery.rs

```diff
@@ -187,159 +233,102 @@ impl NetworkBehaviour for DiscoveryBehaviour {
-
-            list_to_filter.extend(self.kademlia.addresses_of_peer(peer_id));
-            list_to_filter.extend(self.mdns.addresses_of_peer(peer_id));
-
-            // filter private addresses
-            // nodes could potentially report addresses in the private network
-            // which are not actually part of the network
-            if !self.allow_private_addresses {
-                list_to_filter.retain(|addr| match addr.iter().next() {
-                    Some(Protocol::Ip4(addr)) if !IpNetwork::from(addr).is_global() => {
-                        false
-                    }
-                    Some(Protocol::Ip6(addr)) if !IpNetwork::from(addr).is_global() => {
-                        false
-                    }
-                    _ => true,
-                });
-            }
-
-            list.extend(list_to_filter);
-        }
-
-        trace!("Addresses of {:?}: {:?}", peer_id, list);
-
-        list
-    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::{
         DiscoveryBehaviour,
         DiscoveryConfig,
-        KademliaEvent,
+        Event as KademliaEvent,
     };
     use futures::{
         future::poll_fn,
         StreamExt,
     };
     use libp2p::{
-        core,
         identity::Keypair,
         multiaddr::Protocol,
-        noise,
-        swarm::{
-            SwarmBuilder,
-            SwarmEvent,
-        },
-        yamux,
+        swarm::SwarmEvent,
         Multiaddr,
         PeerId,
         Swarm,
-        Transport,
     };
     use std::{
-        collections::{
-            HashSet,
-            VecDeque,
+        collections::HashSet,
+        sync::atomic::{
+            AtomicUsize,
+            Ordering,
         },
-        num::NonZeroU8,
         task::Poll,
         time::Duration,
     };
 
-    /// helper function for building Discovery Behaviour for testing
-    fn build_fuel_discovery(
-        bootstrap_nodes: Vec<Multiaddr>,
-    ) -> (Swarm<DiscoveryBehaviour>, Multiaddr, PeerId) {
-        let keypair = Keypair::generate_secp256k1();
-        let public_key = keypair.public();
+    use libp2p_swarm_test::SwarmExt;
+    use std::sync::Arc;
 
-        let noise_keys = noise::Keypair::<noise::X25519Spec>::new()
-            .into_authentic(&keypair)
-            .unwrap();
-
-        let transport = core::transport::MemoryTransport::new()
-            .upgrade(core::upgrade::Version::V1)
-            .authenticate(noise::NoiseConfig::xx(noise_keys).into_authenticated())
-            .multiplex(yamux::YamuxConfig::default())
-            .boxed();
-
-        let behaviour = {
+    fn build_behavior_fn(
+        bootstrap_nodes: Vec<Multiaddr>,
+    ) -> impl FnOnce(Keypair) -> DiscoveryBehaviour {
+        |keypair| {
             let mut config = DiscoveryConfig::new(
                 keypair.public().to_peer_id(),
                 "test_network".into(),
             );
             config
-                .discovery_limit(50)
+                .max_peers_connected(50)
```

Comment:
Maybe we should move this to a const?

```suggestion
                .max_peers_connected(MAX_PEERS)
```